pkg/block/indexheader/lazy_binary_reader.go

```diff
@@ -23,6 +23,10 @@ import (
 	"github.com/thanos-io/thanos/pkg/objstore"
 )
 
+var (
+	errUnloadedWhileLoading = errors.New("the index-header has been concurrently unloaded")
```

Comment:
`nit` just my pov, do we need here a variable?

Can we just inline the error where it is used? I just checked the codebase it seems we use more the error directly.

But I might be wrong here. :grin: